### PR TITLE
fix(core): typings

### PR DESCRIPTION
**Pull request by kevinawoo**
Fixed some typings preventing building of the `@spinnaker/core` module.
```bash
cd app/scripts/modules/core
yarn lib
```

### CODE REVIEW

File: app/scripts/modules/core/src/utils/timeFormatters.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export function relativeTime(input: any) {
   return thisMoment.isValid() ? thisMoment.fromNow() : '-';
 }
 
-export const fastPropertyTime = memoize((input: any) => {
+export const fastPropertyTime: ((input: any) => string) & _.MemoizedFunction = memoize((input: any) => {
```
Comment (jrsquared) [inline app/scripts/modules/core/src/utils/timeFormatters.ts]: Don't use `_` directly, add `MemoizedFunction` to the import at the top.

File: app/scripts/modules/core/src/instance/instance.write.service.ts [hunk 2]
```diff
@@ -69,7 +69,7 @@ export class InstanceWriter {
     return this.executeMultiInstanceTask(instanceGroups, application, 'rebootInstances', 'Reboot');
   }
 
-  public rebootInstance(instance: IInstance, application: Application, params: any = {}) {
+  public rebootInstance(instance: IInstance, application: Application, params: any = {}): ng.IPromise<ITask> {
```
Comment (jrsquared) [inline app/scripts/modules/core/src/instance/instance.write.service.ts]: add `IPromise` to the angular import line at the top, and don't do `ng.`.

File: app/scripts/modules/core/src/instance/instance.write.service.ts [hunk 3]
```diff
@@ -111,7 +111,7 @@ export class InstanceWriter {
     });
   }
 
-  public registerInstancesWithLoadBalancer(instanceGroups: IMultiInstanceGroup[], application: Application, loadBalancerNames: string[]) {
+  public registerInstancesWithLoadBalancer(instanceGroups: IMultiInstanceGroup[], application: Application, loadBalancerNames: string[]): ng.IPromise<ITask> {
```
Comment (jrsquared) [inline app/scripts/modules/core/src/instance/instance.write.service.ts]: Same, there's a few in this file.

**Comment by jrsquared** [comment]
Good catch, sorry about that, a few changes needed then we can get this merged.

File: app/scripts/modules/core/src/instance/instance.write.service.ts (hunk 2 above)
Comment (kevinawoo) [reply app/scripts/modules/core/src/instance/instance.write.service.ts]: should i change all the other functions too?

Comment (jrsquared) [reply app/scripts/modules/core/src/instance/instance.write.service.ts]: Yes please

**Comment by kevinawoo** [comment]
proof is in the pudding
```bash
~/repos/build/deck/app/scripts/modules/core  [feature/fixed-typings]*
$ git rev-parse HEAD
f5d021456c5c78246cefc099372ba31b50bb4bc4
~/repos/build/deck/app/scripts/modules/core  [feature/fixed-typings]*
$ yarn lib
yarn run v1.3.2
warning package.json: No license field
$ npm run clean && ../../../../node_modules/typescript/bin/tsc && node ../../../../node_modules/webpack/bin/webpack.js

> @spinnaker/core@0.0.111 clean /Users/kevinwoo/repos/build/deck/app/scripts/modules/core
> ../../../../node_modules/rimraf/bin.js lib

^[Starting type checking service...
Using 1 worker with 2048MB memory limit
postcss-colorfix warning:
                ".search .search__input.search__input--focus" uses #66afe9 border-color.
                 Closest variable is --color-text-link
Successfully created style guide at src/styleguide/public/styleguide.html!
✨  Done in 47.59s.
```

**Pull request merged**